File: java/src/main/java/com/example/routeauthorizationtesting/SomeController.java
package com.example.routeauthorizationtesting;

import org.springframework.web.bind.annotation.*;

@RestController
public class SomeController {
    @GetMapping("/user/{id}")
    String getUser(@PathVariable String id) { return "getUser " + id; }

    @PostMapping("/user")
    String postUser() { return "postUser";}

    @PutMapping("/user/{id}")
    String putUser(@PathVariable String id) { return "putUser " + id; }

    @PatchMapping("/user/{id}")
    String patchUser(@PathVariable String id) { return "patchUser " + id; }

    @DeleteMapping("/user/{id}")
    String deleteUser(@PathVariable String id) { return "deleteUser " + id; }

    @GetMapping("/some-untested-route-to-show-failure")
    String untested() { return "nope";}

    @PostMapping("/some-route-without-correct-auth-to-show-failure")
    String wrongAuth() { return "nope";}
}

File: kotlin/src/main/kotlin/com/example/routeauthorizationtesting/SomeController.kt
package com.example.routeauthorizationtesting

import org.springframework.web.bind.annotation.*

@RestController
class SomeController {
    @GetMapping("/user/{id}")
    fun getUser(@PathVariable id: String): String = "getUser $id"

    @PostMapping("/user")
    fun postUser(): String = "postUser"

    @PutMapping("/user/{id}")
    fun putUser(@PathVariable id: String): String = "putUser $id"

    @PatchMapping("/user/{id}")
    fun patchUser(@PathVariable id: String): String = "patchUser $id"

    @DeleteMapping("/user/{id}")
    fun deleteUser(@PathVariable id: String): String = "deleteUser $id"


    @GetMapping("/some-untested-route-to-show-failure")
    fun untested(): String = "nope"

    @PostMapping("/some-route-without-correct-auth-to-show-failure")
    fun wrongAuth(): String = "nope"
}
